sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -5057,6 +5057,22 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-53094: Fix cube-related data quality problem") {
```

Comment:
This is the original author's contribution?

Reply:
Yes, this testcase is from @harris233's https://github.com/apache/spark/pull/51810, I just cleaned it up a bit.